### PR TITLE
Create Nginx image with localy built CLI, Blackbox SDK.

### PR DESCRIPTION
**Pull request by DanLavine**
* Locally builds and adds all CLI OS + arch builds to Nginx
* builds and adds Blackbox SDK
* Update Ansible scripts to deploy the new nginx image

The main motivation for this feature is so deployment options that don't want to rely on Ansible's post docker creation steps can have an Nginx image with all the extra CLIs and SDKs built into it.

**NOTE** This PR will require a new Nginx image for OpenWhisk. I'm not sure where this would be built and published from.

**NOTE** This PR removes the ability to install the CLI from the remote github repo. However, when the CLI is split out of the main OpenWhisk repo it should be easy to pull the remote CLI. All that would need to change is not building the CLI locally and instead, pull the CLI binary/zip files from github into the Nginx image.

### CODE REVIEW

**Comment by csantanapr** [comment]
@markusthoemmes can you take a look at this, what do you think on this approach?

**Comment by markusthoemmes** [comment]
In general, I'm in favor of this approach. It looks sensible to me. I cannot overlook all the sideeffects of how the CLI is built and distributed at this point though.

@houshengbo could you help out here?

File: core/nginx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM nginx:1.11
+
+# make cli directories
+RUN mkdir -p /etc/nginx/cli/go/download/linux/386 && \
+    mkdir -p /etc/nginx/cli/go/download/linux/amd64 && \
+    mkdir -p /etc/nginx/cli/go/download/mac/386 && \
+    mkdir -p /etc/nginx/cli/go/download/mac/amd64 && \
+    mkdir -p /etc/nginx/cli/go/download/windows/386 && \
+    mkdir -p /etc/nginx/cli/go/download/windows/amd64
```
Comment (markusthoemmes) [inline core/nginx/Dockerfile]: Is this needed?

File: core/nginx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+COPY build/windows/386/wsk.exe /etc/nginx/cli/go/download/windows/386/wsk.exe
+COPY build/windows/amd64/wsk.exe /etc/nginx/cli/go/download/windows/amd64/wsk.exe
+
+# copy all cli zip files
+COPY build/linux/386/OpenWhisk_CLI-linux-32bit.tgz /etc/nginx/cli/go/download/linux/386/OpenWhisk_CLI-linux-32bit.tgz
+COPY build/linux/amd64/OpenWhisk_CLI-linux.tgz /etc/nginx/cli/go/download/linux/amd64/OpenWhisk_CLI-linux.tgz
+COPY build/mac/386/OpenWhisk_CLI-mac-32bit.zip /etc/nginx/cli/go/download/mac/386/OpenWhisk_CLI-mac-32bit.zip
+COPY build/mac/amd64/OpenWhisk_CLI-mac.zip /etc/nginx/cli/go/download/mac/amd64/OpenWhisk_CLI-mac.zip
+COPY build/windows/386/OpenWhisk_CLI-windows-32bit.zip /etc/nginx/cli/go/download/windows/386/OpenWhisk_CLI-windows-32bit.zip
+COPY build/windows/amd64/OpenWhisk_CLI-windows.zip /etc/nginx/cli/go/download/windows/amd64/OpenWhisk_CLI-windows.zip
```
Comment (markusthoemmes) [inline core/nginx/Dockerfile]: Couldn't this all be done with one single `COPY`/`ADD`? Same for above cli binaries?

File: core/nginx/build.gradle [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    copy {
+      from "${projectDir}/../../bin/windows/386/OpenWhisk_CLI-windows-32bit.zip"
+      into "${projectDir}/build/windows/386/"
+    }
+    copy {
+      from "${projectDir}/../../bin/windows/amd64/OpenWhisk_CLI-windows.zip"
+      into "${projectDir}/build/windows/amd64/"
+    }
+  }
+}
```
Comment (markusthoemmes) [inline core/nginx/build.gradle]: We should have a look if this is worth being replaced by two `Exec` tasks just doing a `cp` or using a `Copy` task directly. Seems quite error prone.

**Comment by DanLavine** [comment]
Fixed up those issues. I thought there had been an issue when copying files into Docker when the whole path didn't exist. Maybe that was an older docker version though. But I tested the new commits on both Ubuntu and Mac and they were able to build fine.

**Comment by rabbah** [comment]
The SDK is now built externally and the edge router just points to a release. The same could be done for the CLI. This would simplify this PR significantly. Hence I am inclined to close this PR in its current state.

**Comment by csantanapr** [comment]
Yep that's the idea once the CLI is out of the core repo to its own repo then edge would just point to the built binaries on the github release page

**Comment by dgrove-oss** [comment]
Agreed.  Let's close this one and redo with the simpler setup enabled by pulling the artifacts from the github releases of the other repos.

**Pull request closed without merging**